llm/generate/gen_windows.ps1

```diff
@@ -397,15 +397,85 @@ function build_rocm() {
     }
 }
 
+function build_arm64_cpu() {
+    if ((-not "${env:OLLAMA_SKIP_CPU_GENERATE}" ) -and ((-not "${env:OLLAMA_CPU_TARGET}") -or ("${env:OLLAMA_CPU_TARGET}" -eq "cpu"))) {
+        # ARM64 builds have a much better codegen with Clang
+        init_vars
+        $script:buildDir="../build/windows/${script:ARCH}/cpu"
+        $script:distDir="$script:DIST_BASE\cpu"
+        $script:cmakeDefs += @(
+            "-G", "MinGW Makefiles", 
+            "-DCMAKE_C_COMPILER=clang.exe",
+            "-DCMAKE_CXX_COMPILER=clang++.exe",
+            "-DCMAKE_C_FLAGS_INIT=-march=armv8.2-a -fvectorize -ffp-model=fast -fno-finite-math-only",
+            "-DCMAKE_CXX_FLAGS_INIT=-static-libstdc++ -march=armv8.2-a -fvectorize -ffp-model=fast -fno-finite-math-only",
+            "-DCMAKE_EXE_LINKER_FLAGS_INIT=-municode",
```

Comment:
This flag is required to avoid needing this patch: https://github.com/msys2/MINGW-packages/blob/master/mingw-w64-ollama/0003-missing-main.patch